test/client_server_SUITE.erl

```diff
@@ -83,3 +99,24 @@ basic_push(_Config) ->
     ct:pal("Response Body: ~p", [ResponseBody]),
     13 = length(Streams),
     ok.
+
+send_body_opts(_Config) ->
+    {ok, Client} = http2_client:start_link(),
+    RequestHeaders =
+        [
+         {<<":method">>, <<"GET">>},
+         {<<":path">>, <<"/index.html">>},
+         {<<":scheme">>, <<"https">>},
+         {<<":authority">>, <<"localhost:8080">>},
+         {<<"accept">>, <<"*/*">>},
+         {<<"accept-encoding">>, <<"gzip, deflate">>},
+         {<<"user-agent">>, <<"chattercli/0.0.1 :D">>}
+        ],
+
+    ExpectedResponseBody = <<"BodyPart1\nBodyPart2">>,
+
+    {ok, {ResponseHeaders, ResponseBody}} = http2_client:sync_request(Client, RequestHeaders, <<>>),
```

Comment:
It's not really doing that though. For better or worse, this test would pass if the server held back all the response body until it was ready to send it all.

It might mean that we have to add some kind of streaming functionality to the `http2_client`, so that we can see the response coming in pieces. We might have to add some kind of timer delay to the `double_body_handler` to take advantage of that.